ios/brave-ios/Sources/BraveWallet/Extensions/BraveWalletExtensions.swift

```diff
@@ -201,6 +201,16 @@ extension BraveWallet.AccountInfo {
     }
     return parentOrder
   }
+
+  // The account unique key that we use to store for token balance in CD
+  var cacheBalanceKey: String {
+    coin == .btc ? accountId.uniqueKey : address
+  }
+
+  // The account unique key that we use to store for user filter preferences
+  var filterAccountKey: String {
+    coin == .btc ? accountId.uniqueKey : address
+  }
```

Comment:
Spoke in DMs but we should update these to use `accountId.uniqueKey` for uniquely identifying all accounts (and remove these extensions). I believe we will need a migration for the `nonSelectedAccountsFilter` persisted filter value. We can _probably_ get away with clearing balance cache one time instead of migration because balances for the cache are re-fetched when wallet is first opened (ex after next app update with this PR changes), the migrated data would be overwritten quickly anyways (actual migration fine too, assuming balance fetch waits for migration to complete).